File: python/copass-management/src/copass_management/tools/update_agent_tools.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def update_agent_tools(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    slug = str(input["slug"])
    tool_allowlist = [str(x) for x in (input.get("tool_allowlist") or [])]
    agent = await ctx.client.agents.update(
        ctx.sandbox_id,
        slug,
        tool_allowlist=tool_allowlist,
    )
    return {"agent": agent}

File: typescript/packages/management/src/tools/update_agent_tools.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const updateAgentTools: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const slug = String(input.slug);
  const toolAllowlist = Array.isArray(input.tool_allowlist)
    ? input.tool_allowlist.map(String)
    : [];
  const agent = await ctx.client.agents.update(ctx.sandboxId, slug, {
    tool_allowlist: toolAllowlist,
  });
  return { agent };
};
